### ISSUE TITLE
Regarding data upload

### ISSUE THREAD

**Issue by itsthefedora** [openconnectome/Statistical-Connectomics-Class#264]
I can't find any information regarding how to use the suggested "dbinbox" hipster doodad. Is it alright if I just put my data somewhere else online and download it from there?

**Comment by gkiar** [comment]
It was in my original instruction email entitled something like
"Requirements to get an A" ..... You drag and drop, basically.

**Comment by itsthefedora** [comment]
Yeah, but then you want us to download the data in our code, right? Is there a library to download from Dropbox?

**Comment by gkiar** [comment]
Also described in lots of detail in that same email.

**Comment by itsthefedora** [comment]
So, no.

**Comment by itsthefedora** [comment]
Alright.

**Comment by itsthefedora** [comment]
For the future, the syllabus file would probably be a good place to aggregate this kind of information.

**Comment by itsthefedora** [comment]
Also, no, this doesn't work at all as advertised in your email.

I am using urllib.request.urlretrieve in Python 3. It works completely fine if I use a file stored in my own site:
https://itsthefedora.github.io/jhu/statconn/c-elegans-male-1.graphml
and it completely fails if I use the url for your Dropbox site:
https://www.dropbox.com/sh/idt3d0gylplyo31/AABy_BCg3yTxfcEDDo0yOCS_a/c-elegans-male-1.graphml

**Comment by gkiar** [comment]
Multiple people have already submitted and gotten their A. It works.

**Comment by itsthefedora** [comment]
Alright, I got it working. For future reference for people using urllib.request.urlretrieve in Python 3: you **must** include
?dl=1
at the end of the URL, or else it will not work. As in
https://www.dropbox.com/sh/idt3d0gylplyo31/AABy_BCg3yTxfcEDDo0yOCS_a/c-elegans-male-1.graphml?dl=1
Got tipped off to this by the ?dl=0 at the end of the link when you navigate there. Subtle.

**Comment by dlee138** [comment]
For matlab, do we use ?dl=0 or 1?

**Comment by gkiar** [comment]
You can remove the ?dl=# entirely, if you look at my code I show this.

**Comment by itsthefedora** [comment]
Removing it entirely might work in Matlab with wget, but it certainly doesn't work, as I mentioned, in Python 3 with urllib.request.urlretrieve. I can show you an error message if you want.
